generator/constants.ts

```diff
@@ -11,10 +11,25 @@ export const autoRestVerboseOutput = false;
 
 export const schemasBaseUri = 'https://schema.management.azure.com/schemas';
 export const schemasBasePath = path.join(generatorRoot, 'schemas');
-export const generatedSchemasPath = path.join(schemasBasePath, 'common/autogeneratedResources.json');
+export const resourceGroupRootSchema = {
+    file: path.join(schemasBasePath, 'common/autogeneratedResources.json'),
+    jsonPath: 'allOf[1].oneOf'
+};
+export const subscriptionRootSchema = {
+    file: path.join(schemasBasePath, '2018-05-01/subscriptionDeploymentTemplate.json'),
+    jsonPath: 'properties.resources.items.oneOf[0].allOf[1].oneOf'
```

Comment:
The reason that I created a separate "autogeneratedResources" file for resource group resources is because the root schemas sometimes get manual updates, and I wanted to avoid the generation process from interfering with these manual changes.

If this command is run in `generate-all` mode, IIRC it'll remove all references apart from the ones that have been regenerated. Can you check what happens if you run `generate-all`?

Reply:
Yes, I understand. In this PR, I ran `generate-single` for consumption, which has multi-scope resources. You can see the changes in the PR. It's working properly.

When running `generate-all`, it will only remove refs which are listed in whitelist, not all. So it should be fine. Here's the commit running `generate-all`: https://github.com/leni-msft/azure-resource-manager-schemas/commit/ec6e3a683e7eabac3d1f02a4ea3a5743dd22938a

Reply:
Thanks!